cli/src/semgrep/app/scans.py

```diff
@@ -278,6 +278,17 @@ def report_findings(
             for matches_of_rule in matches_by_rule.values()
             for match in matches_of_rule
         ]
+        all_matches.reverse()  # rising filename and line numbers in app
```

Comment:
what's the point of reverse if anyway a few lines below you gonna sort the array?

Reply:
I know it looks unintuitive.  Would a different comment be more useful?  Python 'sorted' maintains the existing sort order of the list, except for the new sort key.  In this code, sorting by severity leaves all "high" severity issues in the original sort order.  The original sort order results in file names appearing in reverse alphabetical order and line numbers decreasing across issues.  I find alphabetical ordering and increasing line numbers more intuitive.

Reply:
Maybe better to be explicit then and fully specify the sorting order rather than relying on obscure internal
of Python functions?

Reply:
Note that "sorted" is a very common python function and sort stability is an important and well-documented "guarantee" (see https://docs.python.org/3/howto/sorting.html).

Nevertheless, agree that in general explicit is better than implicit.  

The order in which we sort issues within a given rule is currently controlled by [get_order_key in rule_match.py](https://github.com/returntocorp/semgrep/blob/develop/cli/src/semgrep/rule_match.py#L228) -- we sort by path, start, end, rule_id, and message.  I believe that sort actually occurs at [core_output.py](https://github.com/returntocorp/semgrep/blob/develop/cli/src/semgrep/core_output.py#L213).  That is the more natural place to reverse the order of the within-rule results, but it will break far more tests and it will affect other output types, whereas I was only seeking to change the ordering for CI upload.

The other place I can do this to make what's happening more explicit is two lines up from the current reverse where we unpack results from the list in the list comprehension [here](https://github.com/returntocorp/semgrep/blob/develop/cli/src/semgrep/app/scans.py#L278).  That could read instead:

        all_matches = [
            match
            for matches_of_rule in sorted(matches_by_rule.values(), reverse=True)
            for match in matches_of_rule
        ]

This incurs some overhead (because we're resorting something that has already been sorted) to the benefit of being more explicit.  Do you prefer this?

Reply:
you can have sorted's key be a tuple for secondary sorting